### DISCUSSION TITLE
Failure to support 3rd party WPF controls is a strategic mistake.

### DISCUSSION THREAD

**Discussion by MarkBrainfire** [AvaloniaUI/Avalonia#12849] [General]
The largest collection of 3rd party controls from venders like Syncfusion, Telerik, Grape City, Infragistics, and many more is WPF controls.

Evidently XPF knows how to access these controls, but Avalonia UI does not.  I realize that the Avalonia team, quite rightly, are using XPF as a key revenue stream.  However, a long view shows that what Avalonia UI needs most now is to grow it's community of users and important projects using the Avalonia UI framework.

Uno Platform just announced version 4.10, which includes support for "Maui Islands", including support for all 3rd party Maui controls, which are substantial.  The ability to access third party controls is so compelling that it tilts the balance towards our friends at Uno Platform.

If possible, Avalonia UI should immediately provide support for 3rd party WPF controls.

**Comment by MikeCodesDotNET** [comment]
Recognising the importance of a robust third-party ecosystem, we strongly urge control vendors to build for Avalonia natively. A growing number of vendors are already in the process of doing so, reinforcing our belief that this is the best long-term approach. 

However, we're not blind to the immediate challenges our customers encounter. To address these, we offer the capability to embed WPF controls into Avalonia applications. While we haven't actively promoted this feature, special XPF licences are available upon request to facilitate this integration. 

**Edit**: We've announced [Hybrid XPF](https://avaloniaui.net/XPF#hybrid), which enables Avalonia developers to use the more than 700 pro controls available for WPF.

**Reply by dbriard** [reply]
May be a good start could be to create a list of what missing controls people need. And there are several useful controls that can be easily ported from WinUI.

**Reply by MikeCodesDotNET** [reply]
We've already seen excellent contributions, like the colour picker port by @robloo. 

The core team is focused on enhancing Avalonia's foundational features, and we're eager for third parties along with the community to enrich the control ecosystem. 

With that said, we welcome pull requests for new controls and see this collaborative approach as vital for Avalonia's growth and success.
